src/controllers/doctor.controller.js

```diff
@@ -0,0 +1,365 @@
+const Doctors = require("../models/doctor.model")
+const Prescription = require("../models/prescription.model")
+const Patients = require("../models/patient.model")
+const Medicine = require("../models/medicine.model")
+const Hospital = require("../models/hospital.model")
+const Bill = require("../models/bill.model")
+
+const bcrypt = require("bcrypt")
+const { newDoctorSchema, 
+        newHospitalSchema, 
+        newMedicineSchema, 
+        addPrescriptionSchema, 
+        addMedicinePrescription, 
+        generatebillSchema,
+        searchPatientSchema
+      } = require("../validators/doctor.validate")
+
+
+exports.newDoctor = async (req, res) =>{
+    try{
+        const result = await newDoctorSchema.validateAsync(req.body)
+        const hospital = await Hospital.findById(result.hospitalId)
+        if(hospital != null || hospital != undefined){
+    
+            const doctor = new Doctors({
+                doctorName: result.doctorName,
+                doctorEmail: result.doctorEmail,
+                doctorPassword: await bcrypt.hash(result.doctorPassword,10),
+                doctorDepartment: result.doctorDepartment,
+                hospitalId: result.hospitalId,
+                hospitalName: hospital.hospitalName
+                
+            })
+            const newDoctor = await doctor.save()
+            return res.json({
+                                status : "Success",
+                                error : false,
+                                message : "Created a new doctor user",
+                                data : newDoctor 
+                            });  
+        }
+    }
+    catch(err){
+        if(err.isJoi === true){
+            return res.json({
+                status : "Failed  validation",
+                error : true,
+                message : "Failed to displayed bill",
+                data : { result: err}
+            })
+        }
+        else{
+        return res.json({
+            status : "Failed",
+            error : true,
+            message : "Failed to find the hospital id",
+            data : { result: err}
+          })
+        }
+    }
+}
```

Comment:
1. Every controller inside a try{}catch(e){} block 
2. Joi for validation and using the payload from Joi for queries
3. Return statement after every function or functional condition
4. res.json() data output formatted in a proper way.
5. error condition in catch block for checking if the error is from validation or something else

These changes are made i every controller file so I am summarizing it in a comment